pkg/util/helper/index.go

```diff
@@ -54,3 +80,39 @@ func IndexerFuncBasedOnLabel(key string) client.IndexerFunc {
 		return []string{val}
 	}
 }
+
+// IndexResourceBinding creates index for ResourceBinding.
+func IndexResourceBinding(ctx context.Context, mgr ctrl.Manager) error {
+	err := mgr.GetFieldIndexer().IndexField(ctx, &workv1alpha2.ResourceBinding{}, IndexNameForResourceBindingDependency, IndexFuncResourceBindingDependencies())
+	if err != nil {
+		klog.Errorf("failed to create index for resource.dependencies, err: %v", err)
+		return err
+	}
+	return nil
+}
+
+// IndexFuncResourceBindingDependencies returns an IndexerFunc used to index resource with the given dependency resource.
+func IndexFuncResourceBindingDependencies() client.IndexerFunc {
```

Comment:
It seems that we don't need this index yet. Should the current pr focus on registering the index in the controller only once?

Reply:
> Should the current pr focus on registering the index in the controller only once?

Maybe can use another pr, focus on registering the index in the controller only once